Bundle/ConfigBundle/Controller/GlobalController.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+namespace Victoire\Bundle\ConfigBundle\Controller;
+
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Route;
+use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Victoire\Bundle\ConfigBundle\Entity\GlobalConfig;
+use Victoire\Bundle\ConfigBundle\Favicon\FaviconGenerator;
+use Victoire\Bundle\ConfigBundle\Form\GlobalConfigType;
+use Victoire\Bundle\CoreBundle\Controller\VictoireAlertifyControllerTrait;
+
+/**
+ * Global config controller.
+ *
+ * @Route("/victoire-dcms/config/global")
+ */
+class GlobalController extends Controller
+{
+    use VictoireAlertifyControllerTrait;
+
+    /**
+     * Method used to edit global config.
+     *
+     * @param Request $request
+     *
+     * @return \Symfony\Component\HttpFoundation\Response
+     *
+     * @Route("/", name="victoire_config_global_edit")
+     */
+    public function editAction(Request $request)
+    {
+        $entityManager = $this->getDoctrine()->getManager();
+        $globalConfig = $entityManager->getRepository(GlobalConfig::class)->find(1) ?: new GlobalConfig();
+        $form = $this->createForm(GlobalConfigType::class, $globalConfig, []);
+        $initialLogo = $globalConfig->getLogo();
+        $form->handleRequest($request);
+
+        if (($submited = $form->isSubmitted()) && $form->isValid()) {
+            $entityManager->persist($globalConfig);
+            $entityManager->flush();
+            if ($initialLogo !== $globalConfig->getLogo() && $globalConfig->getLogo() !== null) {
+                $this->container->get(FaviconGenerator::class)->generate(
+                    $globalConfig,
+                    $this->getParameter('kernel.project_dir').'/faviconConfig.json'
+                );
+            }
+            $this->congrat('victoire.config.global.edit.success');
+
+            return new JsonResponse([
+                'url'     => $this->generateUrl('victoire_core_homepage_show'),
+                'success' => true,
+            ]);
+        }
+
+        return new JsonResponse([
+            'html' => $this->renderView('VictoireConfigBundle:global:edit.html.twig', [
+                'form' => $form->createView(),
+            ]),
+            'success' => !$submited,
```

Comment:
What does it mean? I don't see how these variables are related to each other.

Reply:
I conceed you, it was rather far-fetched and I changed it (https://github.com/Victoire/victoire/pull/1051/commits/e4b02ff6187c10460eaea2f938f9800fd37e5142) even if it was actually working.